subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/ForkingTestClassProcessor.java

```diff
@@ -110,6 +110,10 @@ RemoteTestClassProcessor forkProcess() {
             moduleRegistry.getModule("gradle-native").getImplementationClasspath().getAsURLs(),
             moduleRegistry.getModule("gradle-testing-base").getImplementationClasspath().getAsURLs(),
             moduleRegistry.getModule("gradle-testing-jvm").getImplementationClasspath().getAsURLs(),
+            moduleRegistry.getModule("gradle-testing-junit-platform").getImplementationClasspath().getAsURLs(),
+            moduleRegistry.getExternalModule("junit-platform-engine").getImplementationClasspath().getAsURLs(),
```

Comment:
⭕️ Is this really necessary? I don't see testNG listed here and it works just fine.

Reply:
Well, actually, if a build has dependency `testCompileOnly 'org.testng:testng:6.3.1'`, the result will be:

```
> Task :test FAILED
java.lang.NoClassDefFoundError: org/testng/ITestListener
```
In contrast, if a build has dependency `testCompileOnly 'junit:junit:4.12'`

The result would be:

```
> Task :test FAILED
org.gradle.junit.CategorizedJUnitTest > initializationError FAILED
    java.lang.NoClassDefFoundError
```

Therefore, the dependencies listed here are to guarantee users to see a well-formed warnings. More specifically, all `JUnit Platform` dependencies here can make [this test](https://github.com/gradle/gradle/blob/b11199ea2e14c29bb14a098ba52cf9c4705c4d5e/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy#L55) pass.

It's just an issue of our choice. Both is okay, but I think this is better.